README.md

```diff
@@ -1 +1,297 @@
+- Describe what error driven development is, and how we might do it in Rails
+
+## Framing (10 min)
+
+Let's look at [Facebook](https://www.facebook.com/). What would be missing if Zuck only used HTML, CSS, and JS to build it?
+> We'd have to create a new account every time we open the app.
+> We'd have to repopulate our friend list every time we opened it.
+> Our status would be lost every time we refresh the browser
+
+So, what is a server-side application?
+The generally accepted practice for building a web application is to have your server deliver your homepage and handle saving and loading persistent data, based on simple messages, from the browser
+
+So what is Rails?
+Rails is a heavy duty web server-side framework that follows relatively strict conventions in order to streamline our web development process.
+
+> It is designed to make programming web applications easier by making assumptions about what every developer needs to get started. It makes the assumption that there is the "best" way to do things, and it's designed to encourage that way - and in some cases to discourage alternatives. - Ruby on Rails guide
+
+> Rails is a framework with lots of rules/conventions. Pay attention to the conventions you'll need to follow for rails throughout the week.
+
+Before we get to deep into Rails specifically, we first need to understand the underlying architectural components and the lifecycle of requests.
+
+This lesson will not include writing much (if any) code at all. We are going to take a high-level theoretical approach. It's extremely important to understand the underlying concepts before jumping into such a heavy duty framework such as Rails.
+
+## Request / Response
+
+Open our browsers and type in ESPN to the URL. What's happening?
+Let's identify what is the request, and what is the response?
+
+## Intro to HTTP and REST
+
+### HTTP (5 min)
+
+**Hypertext Transfer Protocol** (HTTP) is a method for encoding and transporting information between a client (such as a web browser) and a web server. HTTP is the primary protocol for transmission of information across the Internet.
+
+A server's job is to respond to HTTP requests. In order to talk about how Rails functions, we need to talk about how HTTP requests work.
+
+Every **HTTP request** consists of a request **method** and **path**. The path is the URL to which the request is being sent. That's pretty familiar. However, your browser always sends that request in a particular way that gives the server a hint as to the "point" of the request. This "particular way" is the method.
+
+Once the request is made, the server grabs all sorts of information and begins to build its **response**. It renders the data on a view (HTML), and then sends it back to the client to see it on the browser.
+
+### HTTP Methods (5 min)
+
+HTTP defines five main methods, each of which corresponds to one of the CRUD functionalities.
```

Comment:
We should instead say `HTTP defines the following methods, each of which corresponds to one of the CRUD functionalities. `